api/src/test/java/org/bahmni/module/events/listener/PatientAdviceTest.java

```diff
@@ -0,0 +1,90 @@
+package org.bahmni.module.events.listener;
+
+import org.bahmni.module.events.model.Event;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.PersonName;
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.springframework.context.ApplicationEventPublisher;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.bahmni.module.events.model.BahmniEventType.PATIENT_CREATED_UPDATED;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@PowerMockIgnore("javax.management.*")
+@PrepareForTest({ ConversionUtil.class })
+@RunWith(PowerMockRunner.class)
+public class PatientAdviceTest {
+	
+	private final PatientAdvice patientAdvice = new PatientAdvice();
+	
+	private ApplicationEventPublisher applicationEventPublisher;
+	
+	@Before
+	public void setUp() {
+		applicationEventPublisher = mock(ApplicationEventPublisher.class);
+		patientAdvice.setApplicationEventPublisher(applicationEventPublisher);
+	}
+	
+	@Test
+	public void shouldVerifyBahmniEventPublishIsCalledGivenNewPatientIsCreated() {
+		Patient newPatient = getPatient();
+		PowerMockito.mockStatic(ConversionUtil.class);
+		PowerMockito.when(ConversionUtil.convertToRepresentation(getPatient(), Representation.FULL)).thenReturn(newPatient);
+		
+		patientAdvice.afterReturning(getPatient(), null, null, null);
+		
+		verify(applicationEventPublisher, times(1)).publishEvent(any(Event.class));
+	}
+	
+	@Test
+	public void shouldVerifyBahmniEventContent() {
+		Patient newPatient = getPatient();
+		
+		PowerMockito.mockStatic(ConversionUtil.class);
+		PowerMockito.when(ConversionUtil.convertToRepresentation(getPatient(), Representation.FULL)).thenReturn(newPatient);
+		
+		patientAdvice.afterReturning(newPatient, null, null, null);
+		
+		ArgumentCaptor<Event> eventArgumentCaptor = ArgumentCaptor.forClass(Event.class);
+		verify(applicationEventPublisher, times(1)).publishEvent(eventArgumentCaptor.capture());
+		
+		Event event = eventArgumentCaptor.getValue();
+		assertEquals(PATIENT_CREATED_UPDATED, event.eventType);
+		assertEquals(newPatient.getUuid(), event.payloadId);
+	}
+	
+	private Patient getPatient() {
+        PersonName name = new PersonName();
```

Comment:
Minor: method not formatted.

Reply:
Weird thing grupreet in Intellij it is showing correct. Not sure how to correct this.

<img width="413" alt="Screenshot 2023-07-04 at 4 21 34 PM" src="https://github.com/Bahmni/openmrs-module-bahmnievents/assets/23061192/6652e3b3-119a-407e-b415-718c5cb8336b">

Reply:
Try VsCode : )

Reply:
If u can't solve this in 10 mins.. u can ignore and mark this comment as resolved.